application/electron/src/services/files/service.file.opener.ts

```diff
@@ -228,6 +260,44 @@ class ServiceFileOpener implements IService {
         });
     }
 
+    private _getFiles(files: string[]): Promise<string[]> {
+        return new Promise((resolve) => {
+            files.forEach((file: string) => {
+                this._listFiles(file);
+            });
+            resolve();
+        });
+    }
+
+    private _listFiles(file: string) {
+        if (fs.lstatSync(file).isFile()) {
+            // File
+            const STATS = fs.statSync(file);
+            return this._files.push({
+                lastModified: STATS.mtimeMs,
+                lastModifiedDate: STATS.mtime,
+                name: this._getName(file),
+                path: file,
+                size: STATS.size,
+                type: 'file',
+            });
+        } else if (!(fs.lstatSync(file).isDirectory())) {
+            // Other
+            return;
+        } else {
+            // Directory
+            fs.readdirSync(file).forEach((subFile: string) => this._listFiles(file + '/' + subFile));
+        }
+    }
```

Comment:
With IPC handlers do not use any IO sync operation. It will block main JS thread. From user point of view we could get frozen UI. Use only async operations. For example instead `fs.statSync`, use `fs.stat`.